docs/spec/spec.md

```diff
@@ -0,0 +1,288 @@
+    region: "ap-south-1"
+};
+
+Client dynamoDBClient = check new (config);
+```
+
+### 2.3 [APIs](#23-apis)
+
+#### [createTable](#createTable)
+
+This API can be used to create a new table in DynamoDB with the specified parameters.
+
+```ballerina
+# Creates table in your account. The CreateTable operation adds a new table to your account. In # an AWS account, table names must be unique within each Region. That is, you can have two # tables with same name if you create the tables in diﬀerent Regions.
+#
+#
+# + tableCreationRequest - The request payload to create a table
+# + return - If success, dynamodb:CreateTableOutput record, else an error
+remote isolated function createTable(CreateTableInput tableCreationRequest) returns TableDescription|error {
+```
+
+#### [deleteTable](#deleteTable)
+
+This API can be used to delete a table in DynamoDB with the given table name.
+
+```ballerina
+# Deletes a table.
+#
+# + tableName - The name of the table to delete
+# + return - If success, dynamodb:DeleteTableOutput record, else an error
+remote isolated function deleteTable(string tableName) returns TableDescription|error {
+```
+
+#### [describeTable](#describeTable)
+
+This API can be used to get the table details in DynamoDB with the given table name.
+
+```ballerina
+# Describes a table.
+#
+# + tableName - The name of the table to delete
+# + return - If success, dynamodb:DescribeTableOutput record, else an error
+remote isolated function describeTable(string tableName) returns TableDescription|error {
+```
+
+#### [listTables](#listTables)
+
+This API can be used to list all the tables relavant to your account.
+
+```ballerina
+# Lists all tables.
+#
+# + return - If success, stream<string, error?>, else an error
+remote isolated function listTables() returns stream<string, error?>|error {
+```
+
+#### [updateTable](#updateTable)
+
+This API can be used to update the entries in the table with the specified parameters.
+
+```ballerina
+# Updates a table.
+#
+# + tableUpdateRequest - The request payload to update a table 
+# + return - If success, dynamodb:UpdateTableOutput record, else an error
+remote isolated function updateTable(UpdateTableInput tableUpdateRequest) returns TableDescription|error {
+```
+
+#### [createItem](#createItem)
+
+This API can be used to add a new item to the table.
+
+```ballerina
+# Creates a new item, or replaces an old item with a new item. If an item that has the same primary key as the new
+# item already exists in the speciﬁed table, the new item completely replaces the existing item. You can perform a
+# conditional put operation (add a new item if one with the speciﬁed primary key doesn't exist), or replace an
+# existing item if it has certain attribute values.
+#
+# + request - The request payload to create an item
+# + return - If success, dynamodb:ItemDescription record, else an error
+remote isolated function createItem(CreateItemInput request) returns ItemDescription|error {
```

Comment:
```suggestion
remote isolated function createItem(ItemCreateInput request) returns ItemDescription|error {
```
Shall we do like this `<subject><verb><input/output>` for record definitions? WDYT?

Reply:
All the comments are fixed in [15cc65b](https://github.com/ballerina-platform/module-ballerinax-aws.dynamodb/pull/31/commits/15cc65b697e9cf7c20776e57d182b98f6cba6684)